### PR TITLE
Add Clear method to SizeConstrainedCache

### PR DESCRIPTION
**Pull request by ganeshvanahalli**
Add useful methods such as `Clear` and `Remove` to `SizeConstrainedCache`. 

Corresponding nitro PR- https://github.com/OffchainLabs/nitro/pull/2754
Part of NIT-2837

### CODE REVIEW

File: common/lru/blob_lru.go [hunk 1]
```diff
@@ -82,3 +82,17 @@ func (c *SizeConstrainedCache[K, V]) Get(key K) (V, bool) {
...
+	defer c.lock.Unlock()
+
+	c.lru.Remove(key)
+}
+
+func (c *SizeConstrainedCache[K, V]) Clear() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.lru.Purge()
```
Comment (magicxyyz) [inline common/lru/blob_lru.go:97]: we should set `c.size = 0` here

**Review by magicxyyz — CHANGES_REQUESTED** [review]
we need to update cache size when removing items or purging the cache

File: common/lru/blob_lru.go [hunk 2]
```diff
@@ -82,3 +82,17 @@ func (c *SizeConstrainedCache[K, V]) Get(key K) (V, bool) {
 
 	return c.lru.Get(key)
 }
+
+func (c *SizeConstrainedCache[K, V]) Remove(key K) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.lru.Remove(key)
```
[the remark this answers is not in the document]
Comment (ganeshvanahalli) [reply common/lru/blob_lru.go]: Thanks! My bad, I overlooked while implementing these methods

**Review by magicxyyz — DISMISSED** [review]
LGTM

**Review by Tristan-Wilson — APPROVED** [review]

**Review by tsahee — APPROVED** [review]
LGTM

**Pull request merged**